### DISCUSSION TITLE
Metals not running when .vsix installed on air gapped computer

### DISCUSSION THREAD

**Discussion by molotch** [scalameta/metals#2542] [Q&A]
Hi,

I'm trying to get a Scala development environment working on an air gapped computer (offline) with only installation files and a JFROG Artifactory available.

Both vscode and the metals (installed using the .vsix) plugin installs fine. It's just that the metals plugin never activate. It doesn't show up in running extensions, it doesn't react when loading a Scala project and vscodes extension host logs don't show anything related to metals.

Any clues on what might be wrong and what I could try?

**Comment by tgodzik** [comment]
Metals is not really suited for offline installation (there is a feature request for that), but you should be able to work around it by specifying coursier mirrors: https://scalameta.org/metals/docs/troubleshooting/proxy.html#mirrors

You should be able to use that alternate Artifactory repo.

**Reply by molotch** [reply]
Thanks, didn't work unfortunately.

Is metals producing any logs one could look at? I did look around but couldn't find any.

**Reply by tgodzik** [reply]
We should have something in `.metals/metals.log` or in the VS Code output panel. The main problem is that we are using minimal coursier boostrap here that downloads main coursier jars directly from Maven Central, which is why we would need the mirrors. It might be possible to just download all separately to the coursier cache.

**Reply by molotch** [reply]
Thanks, the process never gets that far, no .metals folder is created and no log is created, no output in none of the vscode logs either.

I looked at the file list in the bootstrapped coursier file in the scalameta/metals-vscode repo, I guess that's the dependencies to prestore in the coursier cache if I want to try it out?

**Reply by tgodzik** [reply]
I think so, or alternatively you could replace that coursier with a native image or a fully bootstrapped jar.

**Reply by vikas-saxena02** [reply]
@tgodzik did the feature got implemented. we ran into same issue while installing metals from VSIX

**Reply by tgodzik** [reply]
Metals will look for a local coursier if available, but we will still try to download metals server. If it's available in the local cache it should work. What issues are you seeing?

**Reply by vikas-saxena02** [reply]
Hi @tgodzik , as I mentioned in the other thread I have provided the local coursier url to metals but it seems to be unable to find scala.meta.metals.main class.

**Reply by tgodzik** [reply]
Yeah, it looks like it's not downloading the jars correctly 😬

**Reply by vikas-saxena02** [reply]
@tgodzik well that's what I think as well.
I am just wondering if maven can do it as its working without any issues, what's wrong with metals. Any pointers on what could be wrong?

**Reply by tgodzik** [reply]
I would try to use coursier manually to see if it writes anything. It should output all the jars downloaded.

You can also try to run the main class manually and see if there is an issue there.

**Reply by vikas-saxena02** [reply]
Thanks a lot @tgodzik I will give it a rey tomorrow morning ( it's 9 pm in Australia). 
Just curious if I install coursier manually. How do I get metals to use the "local coursier" instead of the bootstrap version?

**Reply by tgodzik** [reply]
We search for it on the PATH variable currently. So you would need to make sure it's executable from anywhere

**Reply by vikas-saxena02** [reply]
@tgodzik thanks for the help. I managed to make some progress and have now run into a new error for which I have posted another discussion
